File: ts/earthbucks-lib/test-jest/domain.test.ts
import { describe, expect, test, beforeEach, it } from "@jest/globals";
import Domain from "../src/domain";

describe("Domain", () => {
  describe("isValidDomain", () => {
    it("should test these known valid/invalid domains", () => {
      expect(Domain.fromIsoStr("earthbucks.com").isValid()).toEqual(true);
      expect(Domain.fromIsoStr("earthbucks.com.").isValid()).toEqual(false);
      expect(Domain.fromIsoStr(".earthbucks.com").isValid()).toEqual(false);
      expect(
        Domain.fromIsoStr("node.node.node.node.earthbucks.com").isValid(),
      ).toEqual(true);
      expect(
        Domain.fromIsoStr(
          "node.node.node.node.node.node.node.node.node.earthbucks.com",
        ).isValid(),
      ).toEqual(false);
    });
  });
});

File: rs/earthbucks_lib/tests/domain_test.rs
use earthbucks_lib::domain::Domain;

#[test]
fn test_is_valid_domain() {
    assert!(Domain::from_iso_str("earthbucks.com".to_string()).is_valid());
    assert!(!Domain::from_iso_str("earthbucks.com.".to_string()).is_valid());
    assert!(!Domain::from_iso_str(".earthbucks.com".to_string()).is_valid());
    assert!(Domain::from_iso_str("node.node.node.node.earthbucks.com".to_string()).is_valid());
    assert!(!Domain::from_iso_str(
        "node.node.node.node.node.node.node.node.node.earthbucks.com".to_string()
    )
    .is_valid());
}
